### PR TITLE
[IMP] top bar: add menu item to insert formula functions

### PR DESCRIPTION
**Pull request by laa-odoo**
## Description:

This commit allows to access and to insert all formula functions
developed in Spreadsheet. The insertion can be done via the
following menu items: 'insert' --> 'function'

Odoo task ID : [2802544](https://www.odoo.com/web#id=2802544&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/spreadsheet/spreadsheet.ts [hunk 1]
```diff
@@ -156,7 +156,7 @@ export class Spreadsheet extends Component<SpreadsheetProps, SpreadsheetChildEnv
     this.sidePanel = useState({ isOpen: false, panelProps: {} });
     this.composer = useState({
       topBarFocus: "inactive",
-      gridFocusMode: "inactive",
+      gridFocusMode: "cellFocus",
```
[the remark this answers is not in the document]
Comment (laa-odoo) [reply src/components/spreadsheet/spreadsheet.ts:159]: The idea is not to have the focus on the right composer, but to have at least one composer activated on both when we open the sheet for the first time.

If you mess with the state prior, (eg click on top bar composer then hit escape to stop editing) and try to insert a formula, then the focus will be on the last composer having the focus (here the top bar composer).
I don't think it's functionally inconvenient. 

This still reflects that the state is not correctly correlated. The state must not be a composition of two substates 'gridFocus' and 'topBarFocus' but one state. I can change this :)

[the remark this answers is not in the document]
Comment (laa-odoo) [reply src/components/spreadsheet/spreadsheet.ts:159]: @rrahir  to do in another pr maybe ?

**Review by LucasLefevre — COMMENTED** [review]
:+1: 
Only minor suggestions

I can't access the last category (web) if my screen is too small and I have to scroll.
The menu is bugging as hell

**Comment by rrahir** [comment]
> +1 Only minor suggestions
> 
> However: I can't access the last category (web) if my screen is too small and I have to scroll. The menu is bugging as hell

there is some issue when you scroll a menu :D

**Comment by rrahir** [comment]
@LucasLefevre I updated the PR with your comments - the "scrolled" menu issue is actually already there with other menus. it seems we render everything when hovering an item with children - which resets the scroll value. THink it can be addressed in another PR

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Pull request closed without merging**